cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -18,6 +18,23 @@
 from cfme.utils.wait import wait_for
 
 
+class EntryPoint(Input):
+    def fill(self, value):
+        current_value = self.value
+        if value == current_value:
+            return False
+        # Clear and type everything
+        self.browser.click(self)
+        self.browser.clear(self)
+        self.browser.send_keys(value, self)
+        self.parent_view.modal.cancel.click()
```

Comment:
Let's not re-implement `widgetastic.widget.TextInput.Fill` here, let's use `super().fill()` here to get through filling of the field, and then add on the `parent_view` access and widget interaction if `TextInput.Fill` returns True.

Reply:
@mshriver 
Can you please explain it for me with example. I am not getting what you are expecting.
Thank you :)

Reply:
So all of the actions in this fill method are the exact same as `TextInput.fill()` method, which is a parent class of this specific Input derived class.

By using `super().fill()`, we'll resolve the `fill()` method from our parent class, and it will execute the fill steps up through calling `self.browser.send_keys()`. Then you can check if `super().fill()` returns True (field was filled in), you can additionally call `self.parent_view.modal.cancel.click()`.

That way any changes to how `TextInput.fill()` works don't have to be copied here as well.

Reply:
Thank you for explanation :)

Reply:
Thanks @mshriver and @digitronik 
Finally, I am able to modify ```fill``` method as shown below:
```

class EntryPoint(Input):
    def fill(self, value):
        if super(EntryPoint, self).fill(value):
            self.parent_view.modal.cancel.click()
            # After changing default path values of 'retirement_entry_point',
            # 'reconfigure_entry_point' or 'provisioning_entry_point';
            # 'Select Entry Point Instance'(pop up) occures which also helps to select these paths
            # using tree structure. Here we have already filled these values through text input.
            # So to ignore this pop up clicking on cancel button is required.
            return True
        return False
```